src/pages/Main/Main.js

```diff
@@ -1,9 +1,109 @@
-import React from 'react';
+import React, { Component } from 'react';
+import Item from '../Shopping/Item/Item';
+import './Main.scss';
+
+export default class Main extends Component {
+  constructor(props) {
+    super(props);
+    this.slideContainer = React.createRef();
+  }
+
+  slideByNumber(e) {
+    this.slideContainer.current.style.transition =
+      'width 2s, height 2s, background-color 2s, transform 1s';
+
+    this.slideContainer.current.style.transform = `translateX(-${
+      576 * e.target.innerHTML
+    }px)`;
+
+    // console.log(this.slideContainer.current.style);
+  }
+  // slideNext() {
+  // console.log(carousel);
+  // let index = 0;
+  // if (index === 0) {
+  //   this.slideContainer.current.style.transform = `translateX(-${576}px)`;
+  //   index = index + 1;
+  // } else {
+  //   this.slideContainer.current.style.transform = `translateX(-${576}px)`;
+  //   index = index + 1;
+  // }
+  // console.log(this.slideContainer.current.style.transform);
+  // console.log(index);
+  // }
+
+  // slidePrevious() {
+  //   // console.log(carousel);
+
+  //   let index = 0;
+
+  //   if (index === 0) {
+  //     this.slideContainer.current.style.transform = `translateX(-${576 * 2}px)`;
+  //     index = index + 1;
+  //   } else {
+  //     this.slideContainer.current.style.transform = `translateX(+${576}px)`;
+  //     index = index + 1;
+  //   }
+  //   console.log(this.slideContainer.current.style.transform);
+  //   console.log(index);
+  // }
 
-class Main extends React.Component {
   render() {
-    return <h1>d</h1>;
+    return (
+      <div className="mainPage">
+        <section className="mainBannerContainer">
+          <div className="mainBanner"></div>
+        </section>
+        <section className="eventBannerContainer">
+          <div className="mainEvent">
+            <div className="slideOverFlow">
+              <div className="slideContentContainer" ref={this.slideContainer}>
+                <div className="eventSlide red"> </div>
+                <div className="eventSlide blue"> </div>
+                <div className="eventSlide yellow"> </div>
+                <div className="eventSlide gr"> </div>
+                <div className="eventSlide wh"> </div>
+                <div className="eventSlide bk"> </div>
+              </div>
+            </div>
+            <div>
+              <div className="slideBtn" onClick={e => this.slideByNumber(e)}>
```

Comment:
- 이대로라면 render 함수가 호출 될때마다 `(e) => this.slideByNumber(e)` 라는 함수가 새로 생성됩니다.
- event 는 자동으로 인자로 들어가기 때문에 직접 따로 명시할 필요가 없습니다!
- 따라서 아래와 같이 작성될 텐데, 이렇게 되면 클래스 메서드로 생성한 slideByNumber 를 재활용하여 불필요한 함수 생성을 방지할 수 있습니다.
```suggestion
              <div className="slideBtn" onClick={this.slideByNumber}>
```

Reply:
네 수정하겠습니다.

Reply:
asyn await 사용해도 되나요?

Reply:
함수 수정했는데 onClick={() => this.slidePrevious()} 이 부분을 지우고 
onClick={this.slidePrevious}이라고 할 경우에 this.setState는 정의되지 않았다 ? 이런식으로 오류가 떠서 
해당부분 수정하려고 했는데 실패해서 일단 () => 문법으로 다시 올렸습니다 ㅜㅜ